frontend/app/controllers/Joiner.scala

```diff
@@ -44,7 +47,16 @@ trait Joiner extends Controller with ActivityTracking {
       request.path,
       Some(CopyConfig.copyDescriptionChooseTier)
     )
-    Ok(views.html.joiner.tierChooser(eventOpt, pageInfo))
+
+    val contentReferer = request.headers.get("referer");
+    val contentAccess = request.getQueryString("membershipAccess")
+
+    if(contentReferer.nonEmpty && contentAccess.nonEmpty) {
+      Ok(views.html.joiner.tierChooser(eventOpt, pageInfo)).withCookies(Cookie("GU_MEM_REFERER", contentReferer.get, secure = true, httpOnly = false))
```

Comment:
I think this cookie _could_ be `httpOnly = true`, and that would be preferable, yes? As it happens, Play actually offers a 'Session' cookie for doing this sort of thing, which has the right defaults:

https://www.playframework.com/documentation/2.3.x/ScalaSessionFlash

I think we should probably be using that for preference.